sdk/resourcemanager/azure-resourcemanager/src/test/java/com/azure/resourcemanager/PrivateLinkTests.java

```diff
@@ -0,0 +1,165 @@
+        Region region = Region.US_EAST;
+
+        StorageAccount storageAccount = azureResourceManager.storageAccounts().define(saName)
+            .withRegion(region)
+            .withNewResourceGroup(rgName)
+            .create();
+
+        Network network = azureResourceManager.networks().define(vnName)
+            .withRegion(region)
+            .withExistingResourceGroup(rgName)
+            .withAddressSpace("10.0.0.0/28")
+            .defineSubnet("default")
+                .withAddressPrefix("10.0.0.0/28")
+                .disableNetworkPoliciesOnPrivateEndpoint()
+                .attach()
+            .create();
+
+        // private endpoint with manual approval
+        PrivateEndpoint privateEndpoint = azureResourceManager.privateEndpoints().define(peName)
+            .withRegion(region)
+            .withExistingResourceGroup(rgName)
+            .withSubnet(network.subnets().get(subnetName))
+            .definePrivateLinkServiceConnection(pecName)
+                .withResource(storageAccount)
+                .withSubResource(PrivateLinkSubResourceName.STORAGE_BLOB)
+                .withManualApproval("request message")
+                .attach()
+            .create();
+
+        Assertions.assertNotNull(privateEndpoint.subnet());
+        Assertions.assertEquals(network.subnets().get(subnetName).innerModel().id(), privateEndpoint.subnet().id());
+        Assertions.assertEquals(1, privateEndpoint.networkInterfaces().size());
+        Assertions.assertEquals(1, privateEndpoint.privateLinkServiceConnections().size());
+        Assertions.assertTrue(privateEndpoint.privateLinkServiceConnections().values().iterator().next().isManualApproval());
+        Assertions.assertEquals("Pending", privateEndpoint.privateLinkServiceConnections().get(pecName).state().status());
+        Assertions.assertEquals("request message", privateEndpoint.privateLinkServiceConnections().get(pecName).requestMessage());
+
+        // approve the connection
+        List<PrivateEndpointConnectionInner> storageAccountConnections = azureResourceManager.storageAccounts().manager().serviceClient().getPrivateEndpointConnections().list(rgName, saName).stream().collect(Collectors.toList());
+        Assertions.assertEquals(1, storageAccountConnections.size());
+        PrivateEndpointConnectionInner storageAccountConnection = storageAccountConnections.iterator().next();
+        Response<PrivateEndpointConnectionInner> approvalResponse = azureResourceManager.storageAccounts().manager().serviceClient().getPrivateEndpointConnections()
+            .putWithResponse(rgName, saName, storageAccountConnection.name(),
+                storageAccountConnection.privateEndpoint(),
+                storageAccountConnection.privateLinkServiceConnectionState().withStatus(PrivateEndpointServiceConnectionStatus.APPROVED),
+                Context.NONE);
+        Assertions.assertEquals(200, approvalResponse.getStatusCode());
```

Comment:
Currently use raw REST API to approve the connection. Later should be switched to fluent API.